### ISSUE TITLE
Locate-only plugins are broken/ignored in 0.20.5 (workaround provided)

### ISSUE THREAD

**Issue by Rush** [systemjs/systemjs#1568]
Sample plugin:
```js
// user for browserConfig only
// import pug files with "sample.pug!" when server is doing the compilation for ".pug.js" requests
exports.locate = function(load) {
  return Promise.resolve(load.address + '.js');
};
```
The result is being assigned here https://github.com/systemjs/systemjs/blob/f7908648d40c3c43d946cef77f2e49adb65c41e0/src/instantiate.js#L158

But since the plugin does not have the `fetch` method it is then being ignored:
https://github.com/systemjs/systemjs/blob/f7908648d40c3c43d946cef77f2e49adb65c41e0/src/instantiate.js#L167

the provided `load.address` is only used when also `.fetch` is provided https://github.com/systemjs/systemjs/blob/f7908648d40c3c43d946cef77f2e49adb65c41e0/src/instantiate.js#L172

**Comment by Rush** [comment]
A working workaround:
```js
exports.fetch = function(load, origFetch) {
  return origFetch(load);
}

exports.locate = function(load) {
  return Promise.resolve(load.address + '.js');
};
```

**Comment by Rush** [comment]
Another issue, previously (in pre-0.20) it was possible to define a plugin without a promise:
```js
exports.locate = function(load) {
  return load.address + '.js';
};
```

Would be nice to have it for compatibility.

**Comment by guybedford** [comment]
Thanks this would be a bug, will aim to fix this.

**Comment by guybedford** [comment]
Fixed in https://github.com/systemjs/systemjs/commit/7d47f1b3fa2652720e49a24aa93ff812b5214f02.

**Comment by Rush** [comment]
@guybedford thanks. What about the first part of the issue where a plugin with `.locate` only is ignored?

**Comment by guybedford** [comment]
@Rush sure I've added https://github.com/systemjs/systemjs/commit/a842efb66472cfe4cb04e790d3e6690225dff88d.

**Comment by guybedford** [comment]
Released in 0.20.7.
